### ISSUE TITLE
Review data checklist 

### ISSUE THREAD

**Issue by Maroyafaied** [ONRR/nrrd#1654]
We need to review that the data checklist still covers everything, one thing we need to add for monthly data is to check the Monthly snapshot.  Also , we need to add a check box to check the download file against the raw file since the download files are automatically generated now.

**Comment by mcharg** [comment]
@Maroyafaied Matt said he should have this done today or tomorrow.

**Comment by Baymatt** [comment]
Here are the changes/additions for the data checklist based on the suggestions above and what I've looked over.

Will add this to all the monthly checklists, 
- [ ] For the current month's data, verify data on monthly snapshot is correct.

Will add in all the downloads file category,
- [ ] Check download file against the raw file to verify both files are the same.

I'm also planning to make a few small changes to the query tool checklist's formatting and options.

**Comment by Maroyafaied** [comment]
@cthomasONRR @Maroyafaied review the checklist and use it to check disbursement data.

**Comment by Maroyafaied** [comment]
@Baymatt Yes, the changes you suggested above sound good, can you also make the following changes;

1. Can you move the Download file category to be on top of the checklist

2.  Add a step at the end of each checklist to check the monthly snapshot against homepage graphs

3. For Disbursements monthly checklist, can you add the following steps under the download file category;

- [ ]  Check  Treasury Fund column is deleted  

- [ ] Check  Hidalgo Caounty is changed to Hidalgo County under county column 

- [ ] Check  "BLM - Permit Processing and Improvement" is changed to to "Lease Processing and Improvement" (BLM) under fund type column

**Comment by Baymatt** [comment]
## Updating Data files

Use the following checklist to check features on the website when updating data files.

**Production Monthly** 

Downloads Page
- [ ] Dates and file size are correct for csv and excel
- [ ] Make sure the downloaded data matches the raw data. Raw data saved here (Insert Teams link to raw file)

Homepage - Yearly Breakdown
- Select a year and verify the amounts and comparisons are correct for the conditions below. Repeat with the most recent year.
    - [ ] Period - Fiscal Year
    - [ ] Period - Calendar Year
    - [ ] Commodity - Oil (bbl)
    - [ ] Commodity - Gas (mcf)
    - [ ] Commodity - Coal (tons)

Homepage - Monthly Breakdown
- Select a month and verify the amounts and comparisons are correct for the conditions below. Repeat with the most recent month. Be sure to verify negative numbers are correct if applicable
    - [ ] Period - Most recent 12 months
    - [ ] Period - Fiscal Year
    - [ ] Period - Calendar Year
    - [ ] Commodity - Oil (bbl)
    - [ ] Commodity - Gas (mcf)
    - [ ] Commodity - Coal (tons)

Homepage - Source File
- [ ] Check source file links are correct.

Query Data
- [ ] One Land Type filter
- [ ] One to three Products
- [ ] Downloads matches filtered data

Monthly Snap Shot
- [ ] For the current month's data, verify data on monthly snapshot is correct. Check homepage graphics against monthly snapshot is correct.

**Production Calendar Year or Fiscal Year**
	
Downloads Page
- [ ] Dates and file size are correct for csv and excel
- [ ] Make sure the downloaded data matches the raw data. Raw data saved here (Insert Teams link to raw file)

Explore Data - Map and Detail Card 
- Select a year and select a state with data, verify information on the detail cards are correct as well as the Production Over Time line graph. Repeat with the latest available year.
    - [ ] Period - Calendar Year
    - [ ] Period - Fiscal Year
    - [ ] With only 2-3 commodities selected
 
- Select a year and select a county with data, verify information on the detail cards are correct as well as the Production Over Time line graph. Repeat with the latest available year.
    - [ ] Period - Calendar Year
    - [ ] Period - Fiscal Year
    - [ ] With only 2-3 commodities selected.
 
- Select a year and select an offshore area, verify information on the detail cards are correct as well as the Production Over Time line graph. Repeat with the latest available year.
    - [ ] Period - Calendar Year
    - [ ] Period - Fiscal Year
    - [ ] With only 2-3 commodities selected.
 
- [ ] Select a state and county without data and verify the expected information is displayed.
- [ ] Add Nationwide Federal Card and verify the expected information is displayed.
- [ ] Add Native American Card and verify the expected information is displayed.

Explore Data - Nationwide Production Summary and Top Nationwide Locations
- Verify breakdowns by production type and source are correct as well nationwide locations for the conditions below.
    - [ ] Not current Calendar Year
    - [ ] Current Calendar Year
    - [ ] Not current Fiscal Year
    - [ ] Current Fiscal Year

Query Data 
- [ ] One Land Type filter
- [ ] One State
- [ ] One Offshore Region
- [ ] One to three Product
- [ ] Downloads matches filtered data

**Revenue Monthly** 	

Downloads Page
- [ ] Dates and file size are correct for csv and excel
- [ ] Make sure the downloaded data matches the raw data. Raw data saved here (Insert Teams link to raw file)

Homepage - Yearly Breakdown
- Select a year and verify the amounts and comparisons are correct for the conditions below. Repeat with the most recent year.
    - [ ] Period - Fiscal Year
    - [ ] Period - Calendar Year
    - [ ] Breakout - Source
    - [ ] Breakout - Revenue Type
    - [ ] Breakout - Commodity
 
Homepage - Monthly Breakdown
- Select a month and verify the amounts and comparisons are correct for the conditions below. Repeat with the most recent month. Be sure to verify negative numbers are correct if applicable
    - [ ] Period - Most recent 12 months
    - [ ] Period - Fiscal Year
    - [ ] Period - Calendar Year
    - [ ] Breakout - Source
    - [ ] Breakout - Revenue Type
    - [ ] Breakout Commodity
 
Homepage - Source File
- [ ] Check source file links are correct.

Query Data
- With the period set to monthly, verify the data is correct and then for the conditions below. 
   - [ ] One Land Type filter
   - [ ] One Revenue Type filter
   - [ ] One State (for Yearly data only)
   - [ ] One Offshore Region
   - [ ] One to three Commodities
   - [ ] Clear All and change grouping
   - [ ] Downloads matches filtered data

Monthly Snap Shot
- [ ] For the current month's data, verify data on monthly snapshot is correct. Check homepage graphics against monthly snapshot is correct.

**Revenue Yearly** 	

Downloads Page
- [ ] Dates and file size are correct for csv and excel
- [ ] Make sure the downloaded data matches the raw data. Raw data saved here (Insert Teams link to raw file)

Homepage - Yearly Breakdown
- Select a year and verify the amounts and comparisons are correct for the conditions below. Repeat with the most recent year.
    - [ ] Period - Fiscal Year
    - [ ] Period - Calendar Year
    - [ ] Breakout - Source
    - [ ] Breakout - Revenue Type
    - [ ] Breakout - Commodity
 
Homepage - Monthly Breakdown
- Select a month and verify the amounts and comparisons are correct for the conditions below. Repeat with the most recent month. Be sure to verify negative numbers are correct if applicable
    - [ ] Period - Most recent 12 months
    - [ ] Period - Fiscal Year
    - [ ] Period - Calendar Year
    - [ ] Breakout - Source
    - [ ] Breakout - Revenue Type
    - [ ] Breakout Commodity
 
Homepage - Source File
- [ ] Check source file links are correct.
 
Explore Data - Map and Detail Card
- Select a year and select a state with data, verify information on the detail cards are correct as well as the Revenue Over Time line graph. Repeat with the latest available year.
    - [ ] Period - Calendar Year
    - [ ] Period - Fiscal Year
    - [ ] With only 2-3 commodities selected
 
- Select a year and select a county with data, verify information on the detail cards are correct as well as the Revenue Over Time line graph. Repeat with the latest available year.
    - [ ] Period - Calendar Year
    - [ ] Period - Fiscal Year
    - [ ] With only 2-3 commodities selected.
 
- Select a year and select an offshore area, verify information on the detail cards are correct as well as the Revenue Over Time line graph. Repeat with the latest available year.
    - [ ] Period - Calendar Year
    - [ ] Period - Fiscal Year
    - [ ] With only 2-3 commodities selected.
 
- [ ] Select a state and county without data and verify the expected information is displayed.
- [ ] Add Nationwide Federal Card and verify the expected information is displayed.
- [ ] Add Native American Card and verify the expected information is displayed.

Explore Data - Nationwide Revenue Summary and Top Nationwide Locations
- Verify breakdowns by revenue type and source are correct as well nationwide locations for the conditions below.
    - [ ] Not current Calendar Year
    - [ ] Current Calendar Year
    - [ ] Not current Fiscal Year
    - [ ] Current Fiscal Year

Query Data
- With a selected fiscal or calendar year range, verify the data is correct and then for the conditions below. 
   - [ ] One Land Type filter
   - [ ] One Revenue Type filter
   - [ ] One State (for Yearly data only)
   - [ ] One Offshore Region
   - [ ] One to three Commodities
   - [ ] Clear All and change grouping
   - [ ] Downloads matches filtered data

**Disbursements Monthly** 

Downloads Page
- [ ] Dates and file size are correct for csv and excel
- [ ] Make sure the downloaded data matches the raw data. Raw data saved here (Insert Teams link to raw file)
- [ ] Check Treasury Fund column is deleted
- [ ] Check Hidalgo Caounty is changed to Hidalgo County under county column
- [ ] Check "BLM - Permit Processing and Improvement" is changed to to "Lease Processing and Improvement" (BLM) under fund type column

Homepage - Yearly Breakdown
- Select a year and verify the amounts and comparisons are correct for the conditions below. Repeat with the most recent year.
- [ ] Breakout - Source 
- [ ] Breakout - Recipient
 
Homepage - Monthly Breakdown
- Select a month and verify the amounts and comparisons are correct for the conditions below. Repeat with the most recent month. Be sure to verify negative numbers are correct if applicable
    - [ ] Period - Most recent 12 months
    - [ ] Period - Fiscal Year
    - [ ] Period - Calendar Year
    - [ ] Breakout - Source
    - [ ] Breakout - Recipient

Query Data
- [ ] One Recipient
- [ ] One Source
- [ ] One State
- [ ] One to three Commodities if applicable
- [ ] Clear All and change grouping
- [ ] Downloads matches filtered data

Monthly Snap Shot
- [ ] For the current month's data, verify data on monthly snapshot is correct. Check homepage graphics against monthly snapshot is correct.

**Disbursements Fiscal Year** 	

Downloads Page
- [ ] Dates and file size are correct for csv and excel
- [ ] Make sure the downloaded data matches the raw data. Raw data saved here (Insert Teams link to raw file)

Explore Data - Map and Detail Card
- With a fiscal year selected, verify the detail card and Disbursements Over Time are correct for the conditions below. Repeat with the current fiscal year.
    - [ ] State with data
    - [ ] State without data
    - [ ] County with data
    - [ ] County without data
    - [ ] Offshore Region
    - [ ] Nationwide Federal Card
    - [ ] Native American Card

Explore Data - Nationwide Revenue Summary and Top Nationwide Locations
- Verify breakdowns by disbursement recipient and source are correct as well nationwide recipients for the conditions below.
    - [ ] Not current Fiscal Year
    - [ ] Current Fiscal Year

Query Data
- [ ] One Recipient
- [ ] One Source
- [ ] One State
- [ ] One to three Commodities if applicable
- [ ] Clear All and change grouping
- [ ] Downloads matches filtered data

**Comment by Maroyafaied** [comment]
@Baymatt This looks great! @cthomasONRR can you review? If everything looks good, we can close this issue.

**Comment by Baymatt** [comment]
Here are the changes based on the checklist that @cthomasONRR updated and that I added to just so we have a list. 
I'm pretty sure we covered everything but if not just let me know.
- Moved Downloads Page checks to the top of each checklist and added a line for downloads vs. raw data.
- Added the additional checks to the Downloads Page checklist for Disbursement Monthly data
- At the end of each monthly checklist, added a Monthly Snapshot checklist.
- Separated Revenue checklist into monthly and yearly and removed the Explore Data section for the Monthly Revenue checklist.
- Removed some Query Data checks from the Production Monthly checklist as they did not apply to the monthly data.
